mlrun/runtimes/sparkjob.py

```diff
@@ -177,19 +177,63 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
         update_in(job, "spec.driver.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.executor.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.deps", self.spec.deps)
+        if "limits" in self.spec.resources:
+            if "cpu" in self.spec.resources["limits"]:
+                update_in(
+                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                )
         if "requests" in self.spec.resources:
             if "cpu" in self.spec.resources["requests"]:
                 update_in(
                     job, "spec.executor.cores", self.spec.resources["requests"]["cpu"]
                 )
-        if "limits" in self.spec.resources:
-            if "cpu" in self.spec.resources["limits"]:
+            if "memory" in self.spec.resources["requests"]:
                 update_in(
-                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                    job,
+                    "spec.executor.memory",
+                    self.spec.resources["requests"]["memory"],
```

Comment:
Are you sure the spark format equals the k8s format ?

Reply:
I'm quite sure it's eventually the same (although it goes through spark params), but why should I care?
I'm using a specific interface for sparkjob and there's no input validation any way.